src/Rules.hs

```diff
@@ -44,7 +45,14 @@ buildLib stage pkg = do
             ways <- interpretInContext context getLibraryWays
             libs <- mapM (pkgLibraryFile . Context stage pkg) ways
             docs <- interpretInContext context $ buildHaddock flavour
-            need $ libs ++ [ pkgHaddockFile context | docs && stage == Stage1 ]
+
+            version <- interpretInContext context (getPkgData Version)
+            let topObj = buildPath context -/- "HS" ++
+                         pkgNameString pkg ++ "-" ++ version ++
+                         (waySuffix (way context) <.> "o")
```

Comment:
Why `|| isStage1Only`? I thought it should be stage-irrelavant ... let's me check out the Makefiles sometime